plugins/weather/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/weather/WeatherResource.java

```diff
@@ -39,65 +36,27 @@
 /**
  * REST Resource for weather
  * 
- * @author Muhammed Kaya, Christian Bräuner
+ * @author Muhammed Kaya, Christian Bräuner, Tim Neumann
  */
 @Path("weather")
 public class WeatherResource implements Resource {
 
 	@Reference
-	private WeatherDarkSkyAPI weatherLogic;
+	private WeatherLogic weatherLogic;
 
 	/**
-	 * get the weather forecast for today
+	 * get the weather report
 	 * 
-	 * @return todays weather forecast
-	 */
-	@GET
-	@Path("now")
-	@Produces(MediaType.APPLICATION_JSON)
-	public WeatherReportNow getWeatherNow() {
-		return this.weatherLogic.getReportNow();
-	}
-
-	/**
-	 * get the weather forecast for today
-	 * 
-	 * @return todays weather forecast
-	 */
-	@GET
-	@Path("today")
-	@Produces(MediaType.APPLICATION_JSON)
-	public WeatherReportDay getWeatherToday() {
-		return this.weatherLogic.getReportToday();
-	}
-
-	/**
-	 * get the weather forecast for the week
+	 * @param locationId
+	 *            The id of the location to get the report for.
 	 * 
-	 * @return this weeks weather forecast
+	 * @return The weather report
 	 */
 	@GET
-	@Path("week")
+	@Path("report")
 	@Produces(MediaType.APPLICATION_JSON)
-	public WeatherReportWeek getWeatherWeek() {
-		return this.weatherLogic.getReportWeek();
-	}
-
-	/**
-	 * set the location for weather forecast
-	 * 
-	 * @param locationId
-	 *            id from the registry entry
-	 */
-	@PUT
-	@Path("setLocation")
-	@Consumes(MediaType.TEXT_PLAIN)
-	public void setLocation(String locationId) {
-		try {
-			this.weatherLogic.setLocation(Integer.parseInt(locationId));
-		} catch (NumberFormatException e) {
-			throw new WebApplicationException("Location not found.", Status.NOT_FOUND);
-		}
+	public WeatherReport getWeatherReport(@QueryParam("id") int locationId) {
```

Comment:
why i can't get the weather for "stuttgart"

Reply:
This will not be possible through the api for now. Could be added in the future.